source/Xunit.Gherkin.Quick/CoreModel/Scenario/Scenario.cs

```diff
@@ -6,17 +6,25 @@
 
 namespace Xunit.Gherkin.Quick
 {
-    internal sealed class Scenario
-    {
-        private readonly ReadOnlyCollection<StepMethod> _steps;
+	internal sealed class Scenario
+	{
+		private readonly Dictionary<string, object> _context;
+		private readonly ReadOnlyCollection<StepMethod> _steps;
 
-        public Scenario(IEnumerable<StepMethod> stepMethods)
-        {
-            _steps = stepMethods != null
-                ? new ReadOnlyCollection<StepMethod>(stepMethods.ToList())
-                : throw new ArgumentNullException(nameof(stepMethods));
-        }
+		public Scenario(IEnumerable<StepMethod> stepMethods)
+		{
+			_context = new Dictionary<string, object>();
+			_steps = stepMethods != null
+				? new ReadOnlyCollection<StepMethod>(stepMethods.ToList())
+				: throw new ArgumentNullException(nameof(stepMethods));
+		}				
 
+		public Scenario(Scenario background, Scenario main)
+		{
+			_context = new Dictionary<string, object>();
+			_steps = background._steps.ToList().Concat(main._steps).ToList().AsReadOnly();		
```

Comment:
Do this concatenation inside `FeatureClass`, when extracting `Scenario`. In general, constructors should not have behavior within as that is not friendly with things such as serialization/persistence/reconstitution. So, to be future-ready, let's try to keep the constructor thin.
This will allow to keep only 1 constructor as well as it will be conceptually the same as before - it's just "extracting a scenario" from feature class, and that automatically extracts the scenario which has steps from the background prepended to it.